### PR TITLE
Fixes various issues related to Tesla API

### PR DESCRIPTION
**Pull request by Bluexin**
See each commit message for details.
#4116 #4143 #3918 (maybe others I didn't find)

### CODE REVIEW

File: src/main/java/mekanism/common/util/CableUtils.java [hunk 1]
```diff
@@ -189,57 +189,30 @@ public static void emit(IEnergyWrapper emitter)
...
-
-					do {
-						double prev = sent;
-						sent += emit_do(emitter, outputtingSides, energyToSend-sent, tryAgain);
-
-						tryAgain = energyToSend-sent > 0 && sent-prev > 0 && i < 100;
-
-						i++;
-					} while(tryAgain);
+					double sent = emit_do(emitter, outputtingSides, energyToSend);
```
Comment (Bluexin) [inline src/main/java/mekanism/common/util/CableUtils.java]: Not 100% sure why this was working the way it was, and I might have overlooked something (all of this is done at 5am).
In my tests, this made everything work as expected (and not as it used to), but I obviously can't check everything.
If you think this should stay (in an adapted way to still fix the issue), please leave your feedback.

**Comment by Bluexin** [comment]
Forgot to mention, the cables do not seem capable of receiving energy from EnderIO capacitor banks and cables, because both of these don't reply to the Tesla outputter capability (this was tested both with and without the capacitor bank in output mode, as well as with and without the cable in PULL mode).
They receive energy from EnderIO generators just fine though (without setting the cable in a special mode).

**Comment by thommy101** [comment]
If you're tackling power issues, there is another Tesla related one. I would fix it myself, but I'm currently busy with school and stuff. The current implementation totally ignores the simulate argument of the tesla capability.
Would be great if you could also tackle that.

**Comment by Bluexin** [comment]
From what I've seen, it takes it into account. I didn't check everything though, only what was related to cables (and their network) as well as energy cubes.
Mind to link me specifically to an example?

PS: I'm also kinda busy, with exams and writing multiple mods for a modpack in addition to fixing existing mods :p

**Comment by thommy101** [comment]
Some places I found it are at the takePower and givePower methods of TeslaIntergration. Changing that would require to add the simulated argument to the upstream call.

**Comment by Bluexin** [comment]
I see now where these are.

@ThatGamerBlue I just saw your "rewrite" fork, but didn't investigate it much yet.
Are your plans to completely rewrite the energy parts (which have gotten damn messy over the years, that's for sure) ?
Do these issues apply to your version, and do you want me to see about porting these?
Finally, do you want me to see about helping to clean things up on your fork?

**Comment by Bluexin** [comment]
While in theory these simulation parameters should work, I can not test it right now.
I made sure everything compiled right on this branch etc but did not test in-game.

**Comment by ThatGamerBlue** [comment]
@Bluexin sorry I couldn't get back to you sooner, yes I plan to rewrite everything, currently there are no issues on my fork because it is very basic (just the metal storage blocks and plastics) at this point but I will probably need help in the future :)

**Comment by roboderpy** [comment]
bump

**Comment by aidancbrady** [comment]
Self-merged, thanks

**Comment by thommy101** [comment]
@aidancbrady I'm still missing the simulated argument which causes unwanted extractions and insertions of energy. Some mods do test if other tiles accept energy with a simulated insertion.

**Comment by roboderpy** [comment]
@aidanebrady why do you do this

**Pull request closed without merging**